### PR TITLE
chore(cypress): Add anchor link tests

### LINKED ISSUE #176: Support for Anchor Links
As a user of the OSCAL Viewer/Editor, I want to be able to quickly navigate to sections of OSCAL documents, including specific controls and control implementations, using anchor links so that I don't have to scroll through often large pages and can bookmark/share specific sections.

### Implementation Notes (and Original Submission)
Within the Catalog Viewer, it would be really helpful if I could link directly to a control within the catalog; for example, if navigating to https://oscal-viewer.msd.easydynamics.com/catalog?url=https://raw.githubusercontent.com/usnistgov/oscal-content/master/nist.gov/SP800-53/rev5/json/NIST_SP-800-53_rev5_catalog.json#SI-3 were to automatically result in "System and Information Integrity" expanded with "SI-3 Malicious Code Protection" focused in the viewport. Perhaps each control could also have a way to copy a link to it (for example, the way that you can do so for headings on the [Microsoft Docs](https://docs.microsoft.com/en-us/windows/wsl/install-on-server#extract-and-install-a-linux-distribution)).

The link should use the `#`/fragment in the URL rather than a querystring parameter for determining where in the page to navigate to, in order to support browsers' native capabilities to do so. This could either be handled by adding an `id` attribute to the headings or placing an `<a>` tag with a name directly above the `<hX>` tags.

I am not sure whether this feature request cleanly maps to the other viewer types.

### PR DESCRIPTION
**Pull request by Bronstrom**
This PR adds Cypress tests ensuring https://github.com/EasyDynamics/oscal-react-library/pull/600 satisfies the desired properties of anchor links. 

These tests ensure that a visited url with a valid hash will navigate to a part of the page and a url contains the correct hash after an anchor link is clicked. Furthermore, anchor links contained within control grouping are used in these test cases as they are the "most buried" anchor links within tab layers, making them the most difficult to test.

Resolves https://github.com/EasyDynamics/oscal-react-library/issues/176

### CODE REVIEW

File: end-to-end-tests/cypress/e2e/OSCAL/OSCAL_Load_Tests.cy.js [hunk 1]
```diff
@@ -102,47 +168,161 @@ describe('Loading Component Definitions', () => {
...
-    cy.contains('Reload').click();
-    cy.contains("Control Groups").should('be.visible');
+    cy.contains("OSCAL Catalog URL")
+      .first()
+      .should("exist")
+      .next()
+      .click()
+      .clear()
+      .type(
+        "https://raw.githubusercontent.com/EasyDynamics/oscal-demo-content/main/catalogs/NIST_SP-800-53_rev4_catalog.json"
```
Comment (tuckerzp) [inline end-to-end-tests/cypress/e2e/OSCAL/OSCAL_Load_Tests.cy.js:254]: Why do extract this into a variable sometimes, but not here?

File: end-to-end-tests/cypress/e2e/OSCAL/OSCAL_Load_Tests.cy.js [hunk 2]
```diff
@@ -1,98 +1,164 @@
...
-    cy.contains('OSCAL System Security Plan URL').first().should('exist').next().click().clear().type(Cypress.env('base_url') + "/oscal/v1/system-security-plans/cff8385f-108e-40a5-8f7a-82f3dc0eaba8")
-    cy.contains('Reload').click()
-    cy.contains(SSP_TITLE_ORIG).should('be.visible')
-    cy.scrollTo('bottom')
-    cy.contains('This is the control implementation for the system.').should('be.visible')
-  })
-  
-  it('displays Enterprise Asset Owners Party', () => {
+describe("Loading system security plans", () => {
+  beforeEach(() => {
```
Comment (tuckerzp) [inline end-to-end-tests/cypress/e2e/OSCAL/OSCAL_Load_Tests.cy.js:7]: I am not a huge fan of `beforeEach` in the tests. I recognize this is repetitive, but assuming each thing we ever test for loading happens in the SSP view is not the best I think.

File: end-to-end-tests/cypress/e2e/OSCAL/OSCAL_Load_Tests.cy.js [hunk 3]
```diff
@@ -1,98 +1,164 @@
...
-    cy.contains("REST Mode").click()
-    cy.contains('OSCAL System Security Plan URL').first().should('exist').next().click().clear().type(Cypress.env('base_url') + "/oscal/v1/system-security-plans/cff8385f-108e-40a5-8f7a-82f3dc0eaba8")
-    cy.contains('Reload').click()
-    cy.contains(SSP_TITLE_ORIG).should('be.visible')
-    cy.scrollTo('bottom')
-    cy.contains('This is the control implementation for the system.').should('be.visible')
-  })
-  
-  it('displays Enterprise Asset Owners Party', () => {
+describe("Loading system security plans", () => {
```
Comment (tuckerzp) [inline end-to-end-tests/cypress/e2e/OSCAL/OSCAL_Load_Tests.cy.js:6]: Moving forward, we need to just fix formatting on all these files in a separate pr. For now, **please do not start arbitrarily fixing style things**. It makes the diff a lot more confusing for no reason. Don't change it, but don't keep doing it.

Comment (Bronstrom) [reply end-to-end-tests/cypress/e2e/OSCAL/OSCAL_Load_Tests.cy.js:6]: To be complete honest I typically use my style formatter to auto format files, which follows in line with our format on our other JS repos. Shouldn't we make file fixes when we touch a file, especially if we touch a large portion of the codebase? About half the tests in this file are related to `OSCALAnchorLinks`.

File: end-to-end-tests/cypress/e2e/OSCAL/OSCAL_Load_Tests.cy.js (hunk 2 above)
Comment (Bronstrom) [reply end-to-end-tests/cypress/e2e/OSCAL/OSCAL_Load_Tests.cy.js:7]: I'll go ahead and reverse this change. I noticed this was done elsewhere in the file.

File: end-to-end-tests/cypress/e2e/OSCAL/OSCAL_Load_Tests.cy.js (hunk 3 above)
Comment (kylelaker) [reply end-to-end-tests/cypress/e2e/OSCAL/OSCAL_Load_Tests.cy.js:6]: Let's not refactor style in this repo until #164 is complete. That ticket will have a massive rework of the linting and style and anything we do until then may be wrong (IDE defaults can be weird).

In general, I like a fix-as-you-go; in `oscal-react-library` we're going to see a lit of diffs like this as TypeScript refactoring happens for example. So I am fine with doing it if there's a really good justification.

In this case, fixing style in a repo with no style rules may create a senseless back-and-forth as each editor does whatever it thinks is best. So if we want to do this, let's solve #164 and fix it all in one go.

@Bronstrom you do not have to revert the changes in order for this PR to be merged; however, let's refrain from doing this again at least until #164 is complete.

File: end-to-end-tests/cypress/e2e/OSCAL/OSCAL_Load_Tests.cy.js [hunk 4]
```diff
@@ -102,47 +191,265 @@ describe('Loading Component Definitions', () => {
...
+      .click()
+      .clear()
+      .type(exampleCompDefUrl);
+    cy.contains("Reload").click();
+    cy.contains("Test Component Definition").should("be.visible");
   });
 
-  it('do not persist after loading a valid component in Editor', () => {
+  it("do not persist after loading a valid component in Editor", () => {
+    sspExampleUrl =
```
Comment (tuckerzp) [inline end-to-end-tests/cypress/e2e/OSCAL/OSCAL_Load_Tests.cy.js:276]:
```suggestion
    const sspExampleUrl =
```

File: end-to-end-tests/cypress/e2e/OSCAL/OSCAL_Load_Tests.cy.js [hunk 5]
```diff
@@ -102,47 +191,265 @@ describe('Loading Component Definitions', () => {
...
+    cy.contains("Reload").click();
+    cy.contains("Yikes").should("be.visible");
   });
 
-  it('do not persist after loading a valid component in Viewer', () => {
-    cy.contains('OSCAL Component URL').first().should('exist').next().click().clear().type("https://raw.githubusercontent.com/EasyDynamics/oscal-demo-content/main/component-definitions/example-component.json");
-    cy.contains('Reload').click();
-    cy.contains('Test Component Definition').should('be.visible');
+  it("do not persist after loading a valid component in Viewer", () => {
+    sspExampleUrl =
```
Comment (tuckerzp) [inline end-to-end-tests/cypress/e2e/OSCAL/OSCAL_Load_Tests.cy.js:249]:
```suggestion
    const sspExampleUrl =
```

**Review by laurelmay — APPROVED** [review]
Assuming these actually pass, I think these do mostly cover the underlying feature and I am good with the content of the tests. Thanks for your work on this.

**Review by tuckerzp — APPROVED** [review]

**Pull request merged**